### DISCUSSION TITLE
Reading struct from BSON errors in block

### DISCUSSION THREAD

**Discussion by BCRARL** [fonsp/Pluto.jl#1383] [Questions about using Pluto]
Although I've defined the struct in the module, I can't read it from a file. No problem on the REPL.
```
begin
     using BSON
end
```
```
struct Reshape
   shape
end
```
```
V = begin
BSON.load(File, @__MODULE__)[:V]
end
```
```
UndefVarError: Reshape not defined
```
Is this expected behavior?

**Comment by BrettKnoss** [comment]
Are you using MongoDB?

**Reply by BCRARL** [reply]
No

**Comment by fonsp** [comment]
I remember seeing this question before, did you also post this question on slack? What was the response?

https://github.com/fonsp/Pluto.jl/issues/301 is related

**Reply by BCRARL** [reply]
I did not ask on Slack, so no response. Maybe somebody else had the same issue.

**Reply by fonsp** [reply]
Ah I found it: https://github.com/fonsp/Pluto.jl/discussions/1380
